dali/pipeline/operators/op_spec.h

```diff
@@ -32,6 +33,65 @@
 
 namespace dali {
 
+namespace detail {
+
+template <typename T, typename S>
+void copy_vector(std::vector<T> &out, const std::vector<S> &in) {
+  out.reserve(in.size());
+  out.clear();
+  for (auto &v : in) {
+    out.emplace_back(v);
+  }
+}
+
+template <typename T>
+void copy_vector(std::vector<T> &out, const std::vector<bool> &in) {
+  out.reserve(in.size());
+  out.clear();
+  for (auto b : in) {
```

Comment:
specialization just to remove the reference from the loop?

Reply:
`vector<bool>` is not an array of bools, but some fancy bit vector - and its iterator's `operator*` does not return a reference :(
I could have used `decltype(auto)` and hope that NVCC9 won't protest.